x-pack/plugins/observability_solution/apm/public/components/app/service_map/index.tsx

```diff
@@ -172,6 +174,10 @@ export function ServiceMap({
     );
   }
 
+  if (status === FETCH_STATUS.SUCCESS) {
+    onPageReady();
+  }
```

Comment:
Don't you want to wrap it within a useEffect? Does it matter if the `onPageReady` is called multiple times?

Reply:
No it doesn't matter, the `onPageReady` should be fired only once when you change the route and the data is loaded. At this point we don't track how long does it take for the updates (ex when you clock refresh)